File: algorithms/sortings/insertionSort.js
function sortByInsertionSort(arr) {
  for (let i = 1; i < arr.length; i++) {
    let j = i - 1;
    let key = arr[i];
    while (j >= 0 && arr[j] > key) {
      arr[j + 1] = arr[j];
      j--;
    }
    arr[j + 1] = key;
  }
}

const nums = [5, 4, 3, 2, 1];

console.log("Unsorted version:", nums);
console.time("Time taken:");
sortByInsertionSort(nums);
console.timeEnd("Time taken:");
console.log("Sorted version:", nums);

File: algorithms/sortings/insertion_sort.py
def insertion_sort(arr):
  for i in range(1, len(arr)):
    j = i - 1
    key = arr[i]
    while j >= 0 and arr[j] > key:
      arr[j + 1] = arr[j]
      j -= 1
    arr[j + 1] = key

# an unsorted array
arr = [10, 20, 15, 12, 11, 5, 8]
insertion_sort(arr)
# sorted array
print(arr)